### DISCUSSION TITLE
ShouldUnregister false and Material's FormControlLabel

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#3033] [Q&A] [answered]
Hello, I'm trying to build a multi step form.
One of the steps will have checkboxes but when I use form control label and click next then back, the checked state disappears even the value becomes true. 

See:
https://codesandbox.io/s/react-hook-form-form-control-label-ll9hy?file=/src/index.js

**Comment by bluebill1049** [comment]
I think for checkbox you should use `render` prop.

**Comment by fadi-george** [comment, marked as answer]
Seems like it, this does seem to work: 
```
 <Controller
    control={control}
    name={name}
    render={(props) => (
      <FormControlLabel
        control={
          <Checkbox
            checked={props.value}
            onChange={(e) => props.onChange(e.target.checked)}
          />
        }
        disabled={disabled}
        label={label}
      />
    )}
  />
```
Not ideal since for non-multi-step forms, inputRef + name works fine w/ checkbox and formcontrollabel w/ checkbox. It'd be nice to not need Controller wrapper

**Reply by bluebill1049** [reply]
yea... ideally more react component do expose input ref in the future.
